sdk/src/compressed_account.rs

```diff
@@ -1,421 +1,215 @@
 use std::ops::{Deref, DerefMut};
 
-use anchor_lang::prelude::{
-    AccountInfo, AnchorDeserialize, AnchorSerialize, ProgramError, Pubkey, Result,
-};
+use anchor_lang::prelude::{AnchorDeserialize, AnchorSerialize, ProgramError, Pubkey, Result};
 use light_hasher::{DataHasher, Discriminator, Hasher, Poseidon};
 use light_utils::hash_to_bn254_field_size_be;
 
 use crate::{
-    address::{derive_address, NewAddressParamsPacked},
-    merkle_context::{
-        pack_merkle_context, MerkleContext, PackedAddressMerkleContext, PackedMerkleContext,
-        RemainingAccounts,
-    },
-    program_merkle_context::unpack_address_merkle_context,
+    account_info::LightAccountInfo,
+    account_meta::LightAccountMeta,
+    address::PackedNewAddressParams,
+    error::LightSdkError,
+    merkle_context::{pack_merkle_context, MerkleContext, PackedMerkleContext, RemainingAccounts},
 };
 
-pub trait LightAccounts: Sized {
-    fn try_light_accounts(
-        inputs: Vec<Vec<u8>>,
-        merkle_context: PackedMerkleContext,
-        merkle_tree_root_index: u16,
-        address_merkle_context: PackedAddressMerkleContext,
-        address_merkle_tree_root_index: u16,
-        remaining_accounts: &[AccountInfo],
-    ) -> Result<Self>;
-    fn new_address_params(&self) -> Vec<NewAddressParamsPacked>;
-    fn input_accounts(
-        &self,
-        remaining_accounts: &[AccountInfo],
-    ) -> Result<Vec<PackedCompressedAccountWithMerkleContext>>;
-    fn output_accounts(
-        &self,
-        remaining_accounts: &[AccountInfo],
-    ) -> Result<Vec<OutputCompressedAccountWithPackedContext>>;
+pub trait LightAccounts<'a>: Sized {
+    fn try_light_accounts(accounts: &'a [LightAccountInfo]) -> Result<Self>;
 }
 
+// TODO(vadorovsky): Implment `LightAccountLoader`.
+
 /// A wrapper which abstracts away the UTXO model.
-pub enum LightAccount<T>
+pub struct LightAccount<'info, T>
 where
-    T: AnchorDeserialize + AnchorSerialize + Clone + DataHasher + Discriminator,
+    T: AnchorDeserialize + AnchorSerialize + Clone + DataHasher + Default + Discriminator,
 {
-    Init(LightInitAccount<T>),
-    Mut(LightMutAccount<T>),
-    Close(LightCloseAccount<T>),
+    /// State of the output account which can be modified by the developer in
+    /// the program code.
+    account_state: T,
+    /// Account information.
+    account_info: LightAccountInfo<'info>,
+    /// Hash of the input account.
+    input_hash: Option<[u8; 32]>,
```

Comment:
duplicate field in `LightAccountInfo` we can probably just use the one in `LightAccountInfo`.